custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -337,6 +343,102 @@ def handle_exception(self, exception, repeat_record):
+            tu_location = SQLLocation.objects.get(location_id=tu_choice)
+        except SQLLocation.DoesNotExist:
+            raise NikshayLocationNotFound(
+                "Location with id {location_id} not found. This is the tu_choice for person with id: {person_id}"
+                .format(location_id=tu_choice, person_id=person_case.case_id)
+            )
+        try:
+            tu_code = tu_location.metadata['nikshay_code']
+        except (KeyError, AttributeError) as e:
+            raise NikshayCodeNotFound("Nikshay codes not found for location with id: {}: {}"
+                                      .format(tu_location.get_id, e))
+        episode_case_date = episode_case_properties.get('date_of_diagnosis', None)
+        if episode_case_date:
+            episode_date = datetime.datetime.strptime(episode_case_date, "%Y-%m-%d").date()
+        else:
+            episode_date = datetime.date.today()
+
+        return {
+            "Stocode": person_locations.sto,
+            "Dtocode": person_locations.dto,
+            "TBUcode": tu_code,
+            "HFIDNO": person_locations.phi,
+            "pname": person_case.name,
+            "fhname": person_case_properties.get('husband_father_name', ''),
+            "age": person_case_properties.get('age', ''),
+            "gender": person_case_properties.get('sex', '').capitalize(),
+            # API for Address with char ',' returns Invalid data format error
+            "Address": person_case_properties.get('current_address', '').replace(',', ''),
+            "pin": person_case_properties.get('current_address_postal_code', ''),
+            "lno": person_case_properties.get('phone', ''),
+            "mno": '0',
+            "tbdiagdate": str(episode_date),
+            "tbstdate": episode_case_properties.get(TREATMENT_START_DATE, str(datetime.date.today())),
+            "Type": disease_classification.get(episode_case_properties.get('pulmonary_extra_pulmonary', ''), ''),
+            "B_diagnosis": episode_case_properties.get('case_definition', ''),
+            "D_SUSTest": episode_case_properties.get('drug_susceptibility_test_status', ''),
+            "Treat_I": episode_case_properties.get('treatment_initiation_status', ''),
+            "usersid": settings.ENIKSHAY_PRIVATE_API_USERS.get(person_locations.sto, ''),
+            "password": settings.ENIKSHAY_PRIVATE_API_PASSWORD,
+            "Source": ENIKSHAY_ID,
+        }
+
+    def handle_success(self, response, payload_doc, repeat_record):
+        # A successful response returns a Nikshay ID like 00001
+        # Failures also return with status code 200 and some message like
+        # Dublicate Entry or Invalid data format
+        # (Dublicate is not a typo)
+        message = get_message_from_xml_response(
+            repeat_record.repeater.url,
+            repeat_record.repeater.operation,
+            response
+        )
+        try:
+            if isinstance(message, basestring) and message.isdigit():
```

Comment:
This won't fail since the response does not return that format. The response has just a 5 digit code in case of success, otherwise it has an error message. You can check failure and success response expected in the tests [here](https://github.com/dimagi/commcare-hq/pull/16149/files#diff-f2705cd281a2209df30e219fc1e93016R63)